commit 59755b58f1a6e2fe1521223a4dcf0d17a51f7181
Author: Lifeng Sun <lifongsun@gmail.com>
Date:   Sun Apr 3 08:22:36 2011 +0800

    fix typo: hierachy -> hierarchy.

diff --git a/fio/HEPEVT_Wrapper.cc b/fio/HEPEVT_Wrapper.cc
index 1f382a8..9906002 100644
--- a/fio/HEPEVT_Wrapper.cc
+++ b/fio/HEPEVT_Wrapper.cc
@@ -186,7 +186,7 @@ namespace HepMC {
 	       << "\n relationships ... but all other information "
 	       << "\n (4-vectors etc) will be correctly transferred."
 	       << "\n In order for HepMC to be able to interpret the mother/"
-	       << "\n daughter hierachy, it MUST be given consistent input."
+	       << "\n daughter hierarchy, it MUST be given consistent input."
 	       << "\n This is one of the design criteria of HepMC: "
 	       << "\n consistency is enforced by the code.";
 	    os << "\nThere is a switch in IO_HEPEVT, set-able using "
