--- a/examples/fio/GNUmakefile.example.in
+++ b/examples/fio/GNUmakefile.example.in
@@ -12,18 +12,16 @@
 #    You may have to change GENSERdir and/or other variables
 #
   HepMCdir             = @prefix@
-  HepMClib             = -L$(HepMCdir)/lib -lHepMC
-  HepMCfiolib          = -L$(HepMCdir)/lib -lHepMCfio
+  HepMClib             = -lHepMC
+  HepMCfiolib          = -lHepMCfio
   GENSERdir            = @GENSERdir@
-  Pythia_LIB	= -L$(GENSERdir)/lib -Wl,-rpath -Wl,$(GENSERdir)/lib \
-                  -lpythia6_403 -lpythia6_403_dumm -lpythia6_403_pdfdumm
-  Herwig_LIB	= -L$(GENSERdir)/lib -Wl,-rpath -Wl,$(GENSERdir)/lib \
-                  -lherwig6_510 -lherwig6_510_dumm -lherwig6_510_pdfdumm
+  Pythia_LIB	= -lpythia6
+  Herwig_LIB	= -lherwig6
 
 ################################################################################ Compiler options
 #
   CXX           = @CXX@
-  F77		= @F77@
+  F77		= gfortran
   INCLUDES 	= -I$(HepMCdir)/include -I$(CLHEPdir)/include
   CXXFLAGS      =  @AM_CXXFLAGS@ @CXXFLAGS@ $(INCLUDES)
 ifeq "$(CXX)" "g++"
--- a/examples/pythia8/GNUmakefile.example.in
+++ b/examples/pythia8/GNUmakefile.example.in
@@ -14,15 +14,13 @@
 #
   HepMCdir      = @prefix@
   HEPMCLOCATION = $(HepMCdir)
-  HepMClib      = -L$(HEPMCLOCATION)/lib -Wl,-rpath -Wl,$(HEPMCLOCATION)/lib \
-                  -lHepMC
-  Pythia_LIB	= -L$(PYTHIA8_HOME)/lib -Wl,-rpath -Wl,$(PYTHIA8_HOME)/lib \
-                  -lpythia8 -llhapdfdummy -lhepmcinterface
+  HepMClib      = -lHepMC
+  Pythia_LIB	= -lpythia8 -lLHAPDF -lhepmcinterface
 
 ################################################################################ Compiler options
 #
   CXX           = @CXX@
-  INCLUDES 	= -I$(HEPMCLOCATION)/include -I$(PYTHIA8_HOME)/include
+  INCLUDES 	= -I/usr/include/pythia8
   CXXFLAGS      =  @AM_CXXFLAGS@ @CXXFLAGS@ -Wshadow -fbounds-check $(INCLUDES)
 ifeq "$(CXX)" "g++"
    FLAGS 	= $(DFLG) -fno-second-underscore $(INCDIR)
