provide dummy hwaend_ function in examples.

--- a/examples/fio/example_MyHerwig.cc
+++ b/examples/fio/example_MyHerwig.cc
@@ -21,6 +21,8 @@
 #include "HepMC/GenEvent.h"
 #include "HepMC/HEPEVT_Wrapper.h"
 
+extern "C" void hwaend_() {}
+
 int main() { 
     //
     //........................................HEPEVT
--- a/examples/fio/testHerwigCopies.cc
+++ b/examples/fio/testHerwigCopies.cc
@@ -13,6 +13,8 @@
 #include "HepMC/CompareGenEvent.h"
 #include "HepMC/HEPEVT_Wrapper.h"
 
+extern "C" void hwaend_() {}
+
 int main() { 
     //
     //........................................HEPEVT
