diff -ur HepMC-2.06.09.orig/fio/Makefile.am HepMC-2.06.09/fio/Makefile.am
--- HepMC-2.06.09.orig/fio/Makefile.am	2011-10-31 20:05:37.000000000 +0100
+++ HepMC-2.06.09/fio/Makefile.am	2012-11-16 16:22:11.373692129 +0100
@@ -10,6 +10,8 @@
 
 lib_LTLIBRARIES = libHepMCfio.la
 
+libHepMCfio_la_LIBADD = $(top_builddir)/src/libHepMC.la
+
 if BUILD_VISUAL
   lib_shared = HepMC.$(SHEXT)
 else
diff -ur HepMC-2.06.09.orig/fio/Makefile.in HepMC-2.06.09/fio/Makefile.in
--- HepMC-2.06.09.orig/fio/Makefile.in	2012-06-05 22:50:44.000000000 +0200
+++ HepMC-2.06.09/fio/Makefile.in	2012-11-16 16:23:07.581987610 +0100
@@ -55,7 +55,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libHepMCfio_la_LIBADD =
+libHepMCfio_la_LIBADD = $(top_builddir)/src/libHepMC.la
 am_libHepMCfio_la_OBJECTS = HEPEVT_Wrapper.lo IO_HEPEVT.lo \
 	IO_HERWIG.lo HerwigWrapper.lo
 libHepMCfio_la_OBJECTS = $(am_libHepMCfio_la_OBJECTS)
